### ISSUE TITLE
reftek: data without header / incomplete packet at the end

### ISSUE THREAD

**Issue by FrankLiu007** [obspy/obspy#3054]
### Problem

I have using obspy to read reftek data, I found "ValueError: invalid literal for int() with base 10: b'  '".  I think it is because of lack of data encoding support for the data. At present, I can't make sure the data encodings  at present. I give an example file of reftek (it is not  a real zip file, it is only reftek130 data).
[091938265_0036EE80.zip](https://github.com/obspy/obspy/files/8564006/091938265_0036EE80.zip)
.

**Comment by megies** [comment]
I think this file might a) be garbled at the end, maybe an interrupted transmission. If I cut off the end until the end of the previous packet, but b) the problem is though that the data does not have an event header/trailer packet. IIRC data packets without a header packet can not be interpreted, not sure though, it's been a while.

**Comment by megies** [comment]
[091938265_0036EE80.crop.reftek.zip](https://github.com/obspy/obspy/files/8564460/091938265_0036EE80.crop.reftek.zip)

This is the file with the incomplete stuff cut off at the end..

you can read it and check out in debugger (e.g. `python -m pdb readit.py'` with just a command to read the file in that `readit.py`)

**Comment by FrankLiu007** [comment]
@megies  It is not a single file, reading almost all the data of the station  failed. It is just a example.  I read the cropped data using obspy.read, but it failed  with:
```  "--> 402         raise TypeError('Unknown format for file %s' % filename)
    403 else:
    404     # format given via argument
    405     format = format.upper()

TypeError: Unknown format for file"
```

**Comment by megies** [comment]
This actually belongs in the https://discourse.obspy.org forum, but since it's already here.. might as well stay here.

If you got sketchy files, you want to manually specify the format, like..

`read("...", format="REFTEK130")` because the file format detection might not auto detect the file and you wanna skip it in this case.

Actually it seems i cropped one byte too few from the file, so its failing the file format detection for RT130.


> I also used the pas2sac tool provided by reftek to convert the data, it works fine.

Well. reftek will have routines that take into account edge cases, but we have an implementation for a regular use case with data packets (DT) properly bounded by event header/trailer (EH/ET) packets, which is the normal case.

I'm not saying you can't read the data, but you are a bit in the wild, since as I said it's not the usual case to lack header/trailer package which has some vital information, otherwise who'd need those header packets anyway? ;-)

```python
from obspy.io.reftek.core import Reftek130
from obspy.io.reftek.packet import _unpack_C0_C2_data, DTPacket

r = Reftek130.from_file('091938265_0036EE80.crop.reftek.zip')
p = DTPacket(r._data[0])
print(p)
_unpack_C0_C2_data(r._data, encoding='C0')
```

Basically you can put it together yourself to read the data, but you are on your own in terms of all the sanity checks that are in the reading routines (e.g. packets could have different data encodings, might not even be sorted correctly in time, etc etc).

**Comment by megies** [comment]
You could either put together code like above to read the data in obspy yourself, or if you have an official conversion tool you might wanna convert it and read the converted data with obspy.
